openmetadata-ui/src/main/resources/ui/src/components/Domain/DomainPage.component.tsx

```diff
@@ -0,0 +1,222 @@
+            } else {
+              return item;
+            }
+          });
+        });
+
+        if (activeDomain?.name !== updatedData.name) {
+          history.push(getDomainPath(response.fullyQualifiedName));
+          fetchDomainList();
+        }
+      } catch (error) {
+        showErrorToast(error as AxiosError);
+      }
+    }
+  };
+
+  const handleDomainDelete = (id: string) => {
+    deleteDomain(id)
+      .then(() => {
+        showSuccessToast(
+          t('server.entity-deleted-successfully', {
+            entity: t('label.domain'),
+          })
+        );
+        setIsLoading(true);
+        // check if the domain available
+        const updatedDomains = domains.filter((item) => item.id !== id);
+        const domainPath =
+          updatedDomains.length > 0
+            ? getDomainPath(updatedDomains[0].fullyQualifiedName)
+            : getDomainPath();
+
+        history.push(domainPath);
+        fetchDomainList();
+      })
+      .catch((err: AxiosError) => {
+        showErrorToast(
+          err,
+          t('server.delete-entity-error', {
+            entity: t('label.domain'),
+          })
+        );
+      })
+      .finally(() => setIsLoading(false));
+  };
+
+  const fetchDomainByName = async (fqn: string) => {
+    setIsMainContentLoading(true);
+    try {
+      const data = await getDomainByName(
+        encodeURIComponent(fqn),
+        'children,owner,parent,experts'
+      );
+      setActiveDomain(data);
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    } finally {
+      setIsMainContentLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchDomainList();
+  }, []);
+
+  useEffect(() => {
+    if (domainFqn) {
+      fetchDomainByName(domainFqn);
+    }
+  }, [domainFqn]);
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  if (!(viewBasicDomainPermission || viewAllDomainPermission)) {
+    return (
+      <ErrorPlaceHolder
+        className="mt-0-important"
+        type={ERROR_PLACEHOLDER_TYPE.PERMISSION}
+      />
+    );
+  }
+
+  if (domains.length === 0 && !isLoading) {
+    return (
+      <ErrorPlaceHolder
+        buttonId="add-domain"
+        className="mt-0-important"
+        doc={GLOSSARIES_DOCS} // Need to replace with domain docs
+        heading={t('label.domain')}
+        permission={createDomainPermission}
+        type={
+          createDomainPermission
+            ? ERROR_PLACEHOLDER_TYPE.CREATE
+            : ERROR_PLACEHOLDER_TYPE.NO_DATA
+        }
+        onClick={handleAddDomainClick}
+      />
+    );
+  }
+
+  return (
+    <PageLayoutV1
+      className="domain-parent-page-layout"
+      leftPanel={<DomainsLeftPanel domains={domains} />}
+      pageTitle={t('label.domain')}>
+      {activeDomain && (
+        <DomainDetailsPage
+          domain={activeDomain}
+          loading={isMainContentLoading}
+          onDelete={handleDomainDelete}
+          onUpdate={handleDomainUpdate}
+        />
```

Comment:
I will be in favour of have the details page code here not sure why we need an additional component for details

Reply:
Wanted to separate out left panel and details page.
If we combine it will be a very big file.